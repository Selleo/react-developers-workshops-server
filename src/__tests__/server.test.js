const request = require('supertest');
const server = require('../server');
const agent = request(server);

describe('Server', () => {
  test('create valid', async () => {
    const response = await agent.post('/todos').send({text: 'Test'});
    expect(response.statusCode).toBe(201);
    expect(JSON.parse(response.text)).toMatchSnapshot();
    await agent.get('/reset');
  });

  test('create invalid', async () => {
    const response = await agent.post('/todos').send({text: ''});
    expect(response.statusCode).toBe(400);
    expect(JSON.parse(response.text)).toMatchSnapshot();
  });

  test('get all', async () => {
    await agent.get('/reset');
    const response = await agent.get('/todos');
    expect(response.statusCode).toBe(200);
    expect(JSON.parse(response.text)).toMatchSnapshot();
  });

  test('get existing', async () => {
    const response = await agent.get('/todos/1');
    expect(response.statusCode).toBe(200);
    expect(JSON.parse(response.text)).toMatchSnapshot();
  });

  test('get not existing ', async () => {
    const response = await agent.get('/todos/123');
    expect(response.statusCode).toBe(404);
    expect(JSON.parse(response.text)).toMatchSnapshot();
  });

  test('update valid', async () => {
    const response = await agent.put('/todos/1').send({text: 'Test'});
    expect(response.statusCode).toBe(200);
    expect(JSON.parse(response.text)).toMatchSnapshot();
  });

  test('update invalid', async () => {
    const response = await agent.put('/todos/1').send({text: ''});
    expect(response.statusCode).toBe(400);
    expect(JSON.parse(response.text)).toMatchSnapshot();
  });

  test('delete', async () => {
    await agent.delete('/todos/1');
    const response = await agent.get('/todos');
    expect(response.statusCode).toBe(200);
    expect(JSON.parse(response.text)).toMatchSnapshot();
    await agent.get('/reset');
  });
});
