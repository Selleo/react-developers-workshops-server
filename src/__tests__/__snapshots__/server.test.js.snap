// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Server create invalid 1`] = `
Object {
  "errors": Array [
    Object {
      "field": "text",
      "location": "body",
      "messages": Array [
        "\\"text\\" is not allowed to be empty",
        "\\"text\\" length must be at least 2 characters long",
      ],
      "types": Array [
        "any.empty",
        "string.min",
      ],
    },
  ],
  "status": 400,
  "statusText": "Bad Request",
}
`;

exports[`Server create valid 1`] = `
Object {
  "id": 4,
  "text": "Test",
}
`;

exports[`Server delete 1`] = `
Array [
  Object {
    "completed": true,
    "id": 2,
    "text": "React",
  },
  Object {
    "completed": false,
    "id": 3,
    "text": "Redux",
  },
]
`;

exports[`Server get all 1`] = `
Array [
  Object {
    "completed": true,
    "id": 1,
    "text": "Json-mock",
  },
  Object {
    "completed": true,
    "id": 2,
    "text": "React",
  },
  Object {
    "completed": false,
    "id": 3,
    "text": "Redux",
  },
]
`;

exports[`Server get existing 1`] = `
Object {
  "completed": true,
  "id": 1,
  "text": "Json-mock",
}
`;

exports[`Server get not existing  1`] = `Object {}`;

exports[`Server update invalid 1`] = `
Object {
  "errors": Array [
    Object {
      "field": "text",
      "location": "body",
      "messages": Array [
        "\\"text\\" is not allowed to be empty",
        "\\"text\\" length must be at least 2 characters long",
      ],
      "types": Array [
        "any.empty",
        "string.min",
      ],
    },
  ],
  "status": 400,
  "statusText": "Bad Request",
}
`;

exports[`Server update valid 1`] = `
Object {
  "id": 1,
  "text": "Test",
}
`;
